File: comparsion/04-concurrency/java/Main.java
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.net.URI;

public class Main {
    public static void main(String[] args) throws Exception {
        HttpClient client = HttpClient.newBuilder().build();
        for (int i = 0; i < 1000; i++) {
            var thread = Thread.ofVirtual().start(() -> {
                HttpRequest request = HttpRequest.newBuilder(URI.create("http://example.com")).build();
                try {
                    HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
                    System.out.println(response.body());
                } catch (Exception e) {
                    e.printStackTrace();
                }
            });
        }
    }
}

File: comparsion/04-concurrency/dotnet/Main.cs
using System.Net.Http;
using System.Threading.Tasks;

public class Main {
    public static async Task Main() {
        using HttpClient client = new HttpClient();
        Task[] tasks = new Task[1000];
        for (int i = 0; i < 1000; i++) {
            tasks[i] = client.GetStringAsync("http://example.com");
        }
        await Task.WhenAll(tasks);
    }
}